pkg/interface/publish/src/js/components/lib/settings.js

```diff
@@ -84,6 +89,112 @@ export class Settings extends Component {
     });
   }
 
+  changeTargetGroup(target) {
+    if (target.groups.length === 1) {
+      this.setState({ targetGroup: target.groups[0] });
+    } else {
+      this.setState({ targetGroup: null });
+    }
+  }
+
+  changeInclusive(event) {
+    this.setState({ inclusive: !!event.target.checked });
+  }
+
+  groupifyNotebook() {
+    const { props, state } = this;
+
+    this.setState({
+      isLoading: true,
+      disabled: true,
+      type: 'Converting'
+    }, (() => {
+      window.api.action("publish", "publish-action", {
+        groupify: {
+          book: props.book,
+          target: state.targetGroup,
+          inclusive: state.inclusive,
+        }
+      }).then(() => this.setState({awaiting: false}));
```

Comment:
We don't use `awaiting` in state for this view, we use `disabled`. (These are used to disable other fields until the poke returns and to in FE mark when to show the `Spinner` component).